docs/book/src/developer/providers/contracts/infra-cluster.md

```diff
@@ -0,0 +1,516 @@
+| [InfraCluster initialization completed]                                   | Yes       |                                                                     |
+| [InfraCluster conditions]                                                 | No        |                                                                     |
+| [InfraCluster terminal failures]                                          | No        |                                                                     |
+| [InfraClusterTemplate, InfraClusterTemplateList resource definition]      | No        | Mandatory for ClusterClasses support                                |
+| [Externally managed infrastructure]                                       | No        |                                                                     |
+| [Multi tenancy]                                                           | No        | Mandatory for clusterctl CLI support                                |
+| [Clusterctl support]                                                      | No        | Mandatory for clusterctl CLI support                                |
+
+Note:
+- `All resources` refers to all the provider's resources "core" Cluster API interacts with;
+  In the context of this page: `InfraCluster`, `InfraClusterTemplate` and corresponding list types
+
+### All resources scope
+
+All resources MUST be namespace-scoped.
+
+### All resources `TypeMeta` and `ObjectMeta` field
+
+All resources MUST have the standard Kubernetes “type metadata” and “object metadata”
+
+### All resources `APIVersion` field value
+
+All resources `APIVersion` is a combination of API group and version.
+
+#### All resources API group
+
+API group MUST be an API group served by the Kubernetes apiserver.
+
+The domain for Cluster API resources is `cluster.x-k8s.io`, and infrastructure providers under the Kubernetes SIGS org
+generally use `infrastructure.cluster.x-k8s.io` as API group.
+
+If your provider uses a different API group, you MUST grant full read/write RBAC permissions for resources in your API group
+to the Cluster API core controllers. The canonical way to do so is via a `ClusterRole` resource with the [aggregation label]
+`cluster.x-k8s.io/aggregate-to-manager: "true"`.
+
+The following is an example ClusterRole for a `FooCluster` resource in the `infrastructure.foo.com` API group:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+    name: capi-foo-clusters
+    labels:
+      cluster.x-k8s.io/aggregate-to-manager: "true"
+rules:
+- apiGroups:
+    - infrastructure.foo.com
+  resources:
+    - fooclusters
+  verbs:
+    - create
+    - delete
+    - get
+    - list
+    - patch
+    - update
+    - watch
+```
+
+Note: The write permissions allow the Cluster controller to set owner references and labels on the InfraCluster” resources;
+write permissions are not used for general mutations of InfraCluster resources.
```

Comment:
I think with ClusterClass, this last sentence is wrong

I would drop this entire Note

Reply:
Also we'll need permissions for the InfraClusterTemplate (IIRC for this one it might only be for ownerRefs)

Reply:
Fixed, PTAL